pkg/network/driver/common.go

```diff
@@ -237,6 +249,38 @@ func (h *NetworkUtilsHandler) NftablesNewChain(proto iptables.Protocol, table, c
 	return nil
 }
 
+func (h *NetworkUtilsHandler) NftablesCreateBaseChains(proto iptables.Protocol) error {
+	if err := h.NftablesNewTable(proto, "nat"); err != nil {
+		return err
+	}
+	if err := h.NftablesNewBaseChain(proto, "nat", "prerouting", "type", "nat", "hook", "prerouting", "priority", "-100"); err != nil {
+		return err
+	}
+	if err := h.NftablesNewBaseChain(proto, "nat", "input", "type", "nat", "hook", "input", "priority", "100"); err != nil {
+		return err
+	}
+	if err := h.NftablesNewBaseChain(proto, "nat", "output", "type", "nat", "hook", "output", "priority", "-100"); err != nil {
+		return err
+	}
+	if err := h.NftablesNewBaseChain(proto, "nat", "postrouting", "type", "nat", "hook", "postrouting", "priority", "100"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (h *NetworkUtilsHandler) NftablesNewBaseChain(proto iptables.Protocol, table, chain string, chainspec ...string) error {
```

Comment:
Maybe it would be simpler to add the optional `chainspec ...string` parameter to the `NftablesNewChain` method.
We wouldn't need to handle the braces and comma, and IMO, the calls would be easier to read:

```go
func (h *NetworkUtilsHandler) NftablesCreateBaseChains(proto iptables.Protocol) error {
	if err := h.NftablesNewTable(proto, "nat"); err != nil {
		return err
	}
	if err := h.NftablesNewChain(proto, "nat", "prerouting", "{ type nat hook prerouting priority -100; }"); err != nil {
		return err
	}
	.
	.
	.
```

There would still be an option to pass individual attributes dynamically, in case we need in the future (the caller would need to add the `{`, `}`, `;` in the arguments).

Reply:
Sounds good